tests/ui/resolve/non-glob-first.rs

```diff
@@ -0,0 +1,21 @@
+// check-pass
+
+mod a {
+    pub trait P {}
+}
+pub use a::*;
+
+mod b {
+    #[derive(Clone)]
+    pub enum P {
+        A
+    }
+}
+pub use b::P;
+
+mod c {
+    use crate::*;
+    pub struct _S(Vec<P>);
+}
+
+fn main() {}
```

Comment:
Is there some specific issue for this test?

Reply:
No specific problem, it just `tests/ui` passed after refactoring but block by this [comment](https://github.com/rust-lang/rust/pull/112659#issuecomment-1593369350). So I reduced it and added it into `tests/ui` after fixed.

btw, It is had some relative with #56593